### PR TITLE
DM-35589: Update to fgcm 3.9.4 which replaces healpy with hpgeom.

### PR DESCRIPTION
**Pull request by erykoff**

### CODE REVIEW

**Review by PaulPrice — COMMENTED** [review]
It looks like more is being done in the first commit than just replacing healpy with hpgeom.

**Comment by PaulPrice** [comment]
No, I was looking at the full diff, and that must have been part of merging master into the LSST branch. Sorry.

**Comment by erykoff** [comment]
There was also removal of unusable code that (a) depended on healpy; (b) depended on code that only worked reliably in py2, so (c) was always turned off.  I finally deleted it.
Anyway, this code isn't formally under DM control, if that matters.

File: fgcm/fgcmMakeStars.py [hunk 1]
```diff
@@ -725,10 +723,12 @@ def makeMatchedStars(self, raArray, decArray, filterNameArray):
...
 
-        theta = (90.0 - self.objCat['dec'][gd])*np.pi/180.
-        phi = self.objCat['ra'][gd]*np.pi/180.
-
-        ipring = hp.ang2pix(self.starConfig['densNSide'], theta, phi)
+        ipring = hpg.angle_to_pixel(
+            self.starConfig['densNSide'],
+            self.objCat['ra'][gd],
+            self.objCat['dec'][gd],
+            nest=False
```
Comment (PaulPrice) [inline fgcm/fgcmMakeStars.py:730]: I'd encourage you to put a comma at the end of the last line, to make it look more black-like. But it's your package.

File: fgcm/fgcmMakeStars.py [hunk 2]
```diff
@@ -804,9 +804,12 @@ def makeReferenceMatches(self, refLoader):
...
                 hasSmatch = False
 
-        ipring = hp.ang2pix(self.starConfig['coarseNSide'],
-                            np.radians(90.0 - self.objIndexCat['dec']),
-                            np.radians(self.objIndexCat['ra']))
+        ipring = hpg.angle_to_pixel(
+            self.starConfig['coarseNSide'],
+            self.objIndexCat['ra'],
+            self.objIndexCat['dec'],
+            nest=False
```
Comment (PaulPrice) [inline fgcm/fgcmMakeStars.py:811]: Ditto.

File: fgcm/fgcmStars.py [hunk 3]
```diff
@@ -1167,13 +1167,11 @@ def plotStarMap(self,mapType='initial'):
...
 
-        import healpy as hp
-        try:
-            from .fgcmPlotmaps import plot_hpxmap
-        except ImportError:
-            self.fgcmLog.info("Map plotting not available.  Sorry!")
-            return
+        # This is not currently used.
+        # FIXME: add skyproj plotting.
+        return
```
Comment (PaulPrice) [inline fgcm/fgcmStars.py:1172]: This probably belongs in a separate commit.

File: fgcm/fgcmStars.py [hunk 4]
```diff
@@ -1167,13 +1167,11 @@ def plotStarMap(self,mapType='initial'):
...
 
+    """
```
Comment (PaulPrice) [inline fgcm/fgcmStars.py:1174]: I don't think disabling code like this is complying with LSST code standards.

**Review by PaulPrice — APPROVED** [review]

File: fgcm/fgcmStars.py (hunk 4 above)
Comment (erykoff) [reply fgcm/fgcmStars.py:1174]: Probably not, but I wanted to keep it around to when I reenable it using `skyproj`.
